### PR TITLE
Esri Open Source Community Code of Conduct

### PR DESCRIPTION
**Pull request by ChrisFrenchPDX**
Initial draft for an Esri Open Source Community Code of Conduct

### CODE REVIEW

File: COMMUNITY_CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+## Community Expectations
+- Feel free to share your ideas, but also listen to others
+- To be professional, kind, and considerate in technical discussions and disagreements
+- Be respectful of personal boundaries and preferences
+- Refrain from demeaning, harassing, aggressive, or intimidating communications and behavior
+- Be respectful of all project contributors and maintainers
+- Report inappropriate behavior
+- Contributors must have the legal right to contribute source code and content under the Apache 2.0 license
+
+## Inappropriate behavior—harassment, aggression, and intimidation
```
Comment (mikewilburn) [inline COMMUNITY_CODE_OF_CONDUCT.md]: I don't think we should limit this section heading to "harassment, aggression, and intimidation" as your second paragraph points out. I think simply "Inappropriate behavior" is fine.

File: COMMUNITY_CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+- Harassment, aggression, and intimidation are hurtful and interfere with other people’s experience and participation in our community. These behaviors are inappropriate and unacceptable.
+
+- Inappropriate behavior can take many forms on GitHub, Email, Slack, and other communications, such as: slurs; derogatory jokes and statements; foul or obscene language; stalking; graphic or derogatory pictures; drawings, or cartoons; _ad hominem_ or personal attacks and insults; unwanted or offensive letters or poems; offensive email or voicemail messages; or social media postings.
+
+## Consequences of inappropriate behavior
+
+- Inappropriate behavior will not be tolerated. If asked to stop behaving inappropriately, you should comply immediately
+- Esri does not tolerate harassment or personal threats
+- Violations of this Community Code of Conduct may result in disqualifications in Esri Events, including conferences. [See the Esri Code of Conduct for Events and Conferences](https://www.esri.com/en-us/about/events/code-of-conduct).
```
[the remark this answers is not in the document]
Comment (mikewilburn) [reply COMMUNITY_CODE_OF_CONDUCT.md]: Some of these aren't consequences and just restate that there _will_ be consequences. I propose that we add this as a course of action:
https://help.github.com/articles/blocking-a-user-from-your-organization/

**Comment by jf990** [comment]
Can we consider naming the file CODE_OF_CONDUCT.md ? That seems to be the npm/OSS standard. Someone even has a [CoC generator](https://github.com/WeAllJS/weallbehave)

**Comment by ChrisFrenchPDX** [comment]
Ok, I've made a few changes and edits based on everyone's suggestions:

- @jf990 I deleted the `COMMUNITY_CODE_OF_CONDUCT.md` file and moved the contents to `CODE_OF_CONDUCT.md`
- @mikewilburn  and @jgravois I added a version/data to the doc, moved some the contributing info the the `README.md` file, clarified consequences for violations, and restructured the file re the suggestions

**Comment by jgravois** [comment]
@ChrisFrenchPDX i have a few more copy editing nitpicks. can i just push them up in a commit of my own?

**Comment by ChrisFrenchPDX** [comment]
@jgravois Sure!

**Comment by ChrisFrenchPDX** [comment]
Ok, cool. The changes look good to me - I just changed a few punctuation typos.

**Comment by mikewilburn** [comment]
This is looking real solid to me. Happy with the latest changes!

**Pull request merged**